pkg/medusa/reconcile.go

```diff
@@ -182,6 +189,100 @@ func UpdateMedusaMainContainer(dcConfig *cassandra.DatacenterConfig, medusaSpec
 	}
 }
 
+func medusaMainContainer(dcConfig *cassandra.DatacenterConfig, logger logr.Logger) (int, bool, *corev1.Container) {
+	medusaContainerIndex, found := cassandra.FindContainer(dcConfig.PodTemplateSpec, "medusa")
+	medusaContainer := &corev1.Container{Name: "medusa"}
+	if found {
+		logger.Info("Found medusa container")
+
+		medusaContainer = dcConfig.PodTemplateSpec.Spec.Containers[medusaContainerIndex].DeepCopy()
+	}
+	return medusaContainerIndex, found, medusaContainer
+}
+
+// Creates a deployment for a standalone Medusa pod which will be used by the operator to interact directly with the storage backend.
+// It allows such interactions before any Cassandra pod is created, to enable performing a restore on the first startup.
+func StandaloneMedusaDeployment(dcConfig *cassandra.DatacenterConfig, medusaSpec *api.MedusaClusterTemplate, clusterName string, volumes []medusaVolume, additionalVolumes []medusaAdditionalVolume, namespace string, logger logr.Logger) *appsv1.Deployment {
+	_, _, medusaContainer := medusaMainContainer(dcConfig, logger)
+
+	medusaDeployment := &appsv1.Deployment{
```

Comment:
I might be wrong here since it isn't quite clear where medusaContainer comes from (in the worst case it seems to be an empty newly allocated Container from medusaMainContainer). So apologies if this is a silly question...

Do we have liveness and readiness checks for this container?